lib/stat_tracker.rb

```diff
@@ -121,19 +121,78 @@ def games_won_per_team_per_season(season_id)
   def shots_to_goals_ratio_per_team_per_season(season_id)
     total_shots_per_team_per_season(season_id).merge(total_goals_per_team_per_season(season_id)){|team_id, shots, goals| (shots.to_f / goals).round(2)}
   end
+
   def total_goals_per_team_per_season(season_id)
     @game_teams.reduce(Hash.new(0)) do |result, game_team|
       result[game_team.team_id] += game_team.goals if game_team.season == season_id
       result
     end
   end
+
   def total_shots_per_team_per_season(season_id)
     @game_teams.reduce(Hash.new(0)) do |result, game_team|
       result[game_team.team_id] += game_team.shots if game_team.season == season_id
       result
     end
   end
 
+  def total_games_per_team_per_season(season_id)
+    @game_teams.reduce(Hash.new(0)) do |result, game_team|
+      result[game_team.team_id] += 1 if game_team.season == season_id
+      result
+    end
+  end
+
+  def games_won_per_team_for_season(season_id)
+    @game_teams.reduce(Hash.new(0)) do |result, game_team|
+      result[game_team.team_id] += 1 if game_team.season == season_id &&  game_team.result == "WIN"
+      result
+    end
+  end
+
+  def total_games_per_team_for_season(season_id)
+    @game_teams.reduce(Hash.new(0)) do |result, game_team|
+      result[game_team.team_id] += game_team.goals if game_team.season == season_id
+      result
+    end
+  end
```

Comment:
Hey! Is this a repeat of total_games_per_team_per_season(season_id) above on line 139?

Reply:
Good catch! Sorry about that. They're all starting to blur together!

Reply:
They really are!